File: app/src/main/java/nz/co/martinpaulo/wordjambalaya/UnknownWords.java
package nz.co.martinpaulo.wordjambalaya;

import android.content.Context;

import java.util.ArrayList;

/**
 * Created by martin paulo on 29/07/2014.
 */
public class UnknownWords {

    private static UnknownWords instance;
    private Context context;


    private ArrayList<UnknownWord> unknownWords;

    private UnknownWords(Context context) {
        this.context = context;
        unknownWords = new ArrayList<UnknownWord>();
    }

    public static UnknownWords getInstance(Context context, int numberOfWords) {
        if (instance == null) {
            instance = new UnknownWords(context);
        }
        for (int i = instance.getWords().size(); i < numberOfWords; i++) {
            final UnknownWord word = new UnknownWord();
            instance.unknownWords.add(word);
        }
        return instance;
    }

    public ArrayList<UnknownWord> getWords() {
        return unknownWords;
    }
}

File: app/src/main/java/nz/co/martinpaulo/wordjambalaya/UnknownWords.kt
package nz.co.martinpaulo.wordjambalaya

/**
 * Essentially a singleton wrapper around the list of unknown words
 * Created by martin paulo on 29/07/2014.
 */
class UnknownWords private constructor() {
    val words: ArrayList<UnknownWord> = ArrayList()

    companion object {
        private var instance: UnknownWords? = null


        fun getInstance(numberOfWords: Int): UnknownWords? {
            if (instance == null) {
                instance = UnknownWords()
            }
            for (i in instance!!.words.size until numberOfWords) {
                val word = UnknownWord()
                instance!!.words.add(word)
            }
            return instance
        }
    }
}
